src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php

```diff
@@ -402,6 +403,15 @@ public function get_item( $request ) {
 			return $revision;
 		}
 
+		if ( $parent->ID !== $revision->post_parent ) {
+			return new WP_Error(
```

Comment:
This seems fine from a code perspective. How could it explode for consumers of the API?

Reply:
It won't bode well for folks who, for whatever reason, probably unintentionally, are expecting values back from a mismatched combination of parentId and revisionId. 

The argument for dismissing this concern is that, in such as case, the URL is not representational of the resource, and so it is not RESTful. It's probably worth a dev note on the Make Core blog.

How it could explode in another scenarios? I'm not sure. We're verifying the existence of both parent and revision posts, and their post types. I suppose if the expected properties, ID and post_parent, are unavailable but that would be a deeper issue I believe.

Reply:
Yeah, let's just call it a bug and make sure we mention the change of behavior in a Dev Note

Reply:
I think this should be moved to the permission check.

Reply:
@jonnynews Can you explain further? I think the current location is appropriate because it's a data discrepancy check, not a security check. There are data discrepancy checks immediately above too.

Reply:
It's it a security check, as the current user might not have access.

Reply:
> as the current user might not have access

I agree prima facie with @danielbachhuber here that it's data mismatch check, but I'd like to understand the concerns. Could you please elaborate a little more? 

Do you mean access to the revision itself? There's a test in `get_items_permissions_check()` to ensure the user has edit rights to the parent. Here we're verifying whether the revision matches the parent, not explicitly whether the user has access to the revision entity.

Unless you think we should add a `current_user_can` for `$revision->ID`  in `get_items_permissions_check()`?

Reply:
Before we go ahead and merge this, it'd be helpful to have some clarification on your comment if you have time @jonnynews 

Otherwise we can always do a follow up PR with any constructive feedback.

Reply:
Sorry, this is my other account. Looking at the rest of the revision controller, it seems that other checks are like this. So I am going to dismiss.